File: leetcode/剑指/剑指.9.两个栈实现一个队列.go
type CQueue struct {
    s1 []int;
    s2 []int;
}


func Constructor() CQueue {
    var queue CQueue;
    return queue;
}


func (this *CQueue) AppendTail(value int)  {
    this.s1 = append(this.s1, value);
}


func (this *CQueue) DeleteHead() int {
    if(len(this.s2) == 0) {
        for {
            var l1 = len(this.s1);
            if(l1 == 0) {
                break;
            }

            this.s2 = append(this.s2, this.s1[l1 - 1]);
            this.s1 = this.s1[:l1 - 1];
        }
    }
    var l2 = len(this.s2)
    if(l2 == 0) {
        return -1
    } else {
        var ret int = this.s2[l2 - 1];
        this.s2 = this.s2[:l2-1];
        return ret;
    }
}


/**
 * Your CQueue object will be instantiated and called as such:
 * obj := Constructor();
 * obj.AppendTail(value);
 * param_2 := obj.DeleteHead();
 */

File: leetcode/剑指/剑指.9.两个栈实现一个队列.py
class CQueue:
    
    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def appendTail(self, value: int) -> None:
        self.stack1.append(value)

    def deleteHead(self) -> int:
        if not self.stack1 and not self.stack2:
            return -1
        if self.stack2:
            return self.stack2.pop()
        while self.stack1:
            num = self.stack1.pop()
            self.stack2.append(num)
        return self.stack2.pop()


# Your CQueue object will be instantiated and called as such:
# obj = CQueue()
# obj.appendTail(value)
# param_2 = obj.deleteHead()